### PR TITLE
FISH-6768 display additionalProperties if specified

### PR DESCRIPTION
**Pull request by aubi**
## Description
Adding support of additionalProperties to the generated OpenApi documentation.
If additionalProperties is not specified in the `@Schema` annotation, it is supposed to be True, but it is not mentioned in the displayed documentation (it's the default value).
If explicitly specified, it is always displayed, either true or false.
Deploy the reproducer attached to the Jira ticket to Payara and open `http://localhost:4848/openapi` in browser.

The key parts of the source:
```
@Schema(description = "no additionalProperties defined, shouldn't appear in OpenApi description")
public static class WithoutAdditionalProperties {

@Schema(additionalProperties = Schema.True.class, description = "additionalProperties defined as true, should appear as 'additionalProperties: true'")
public static class WithAdditionalProperties {

@Schema(additionalProperties = Schema.False.class, description = "additionalProperties defined as false, should appear as 'additionalProperties: false'")
public static class WithNoAdditionalProperties {
```
This output should be displayed (unrelated parts removed):
```
    WithAdditionalProperties:
      description: "additionalProperties defined as true, should appear as 'additionalProperties:\
        \ true'"
      additionalProperties: true
    WithNoAdditionalProperties:
      description: "additionalProperties defined as false, should appear as 'additionalProperties:\
        \ false'"
      additionalProperties: false
    WithoutAdditionalProperties:
      description: "no additionalProperties defined, shouldn't appear in OpenApi description"
```

Last tiny change -- the version in the first line is changed to 3.1.0.

### New tests
3 new tests in appserver/payara-appserver-modules/microprofile/openapi/src/test/java/fish/payara/microprofile/openapi/test/app/application/SchemaExampleTest.java

### Testing Performed
Unit tests, test of reproducer and 4848/openapi

### Testing Environment
Linux, OpenJDK 11

### Info for Reviewers
I split the changes to 3 commits, so it's easier to review. The first one is functional, then change of generated version, then removal of deprecation warnings. The last commit has no functional impact.

### CODE REVIEW

**Review by luiseufrasio — CHANGES_REQUESTED** [review]
The reproducer.zip file is empty

**Comment by aubi** [comment]
fixed

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
     DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
 
-    Copyright (c) 2011-2015 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2011-2023 Oracle and/or its affiliates. All rights reserved.
```
Comment (luiseufrasio) [inline pom.xml:6]: 2011-2015 refers to Oracle versioning
Suggestion: add a comment bellow with 2023 regarding Payara versioning

**Review by luiseufrasio — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (aubi) [reply pom.xml]: Fixed, thank you!

**Review by luiseufrasio — APPROVED** [review]
LGTM!

**Comment by aubi** [comment]
Jenkins test please

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/BaseProcessor.java [hunk 2]
```diff
@@ -76,6 +76,8 @@ public OpenAPI process(OpenAPI api, OpenApiConfiguration config) {
 
         // Set the OpenAPI version if it hasn't been set
         if (api.getOpenapi() == null) {
+            // FIXME: change to 3.1.0 when MP TCK is upgraded
+            // api.setOpenapi("3.1.0");
             api.setOpenapi("3.0.0");
```
Comment (breakponchito) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/BaseProcessor.java:81]: Do we need to wait the upgrade? I saw that you changed version on other parts

File: pom.xml [hunk 3]
```diff
@@ -215,7 +215,7 @@
         <microprofile-healthcheck.version>4.0.1</microprofile-healthcheck.version>
         <microprofile-metrics.version>4.0</microprofile-metrics.version>
         <microprofile-rest-client.version>3.0.1</microprofile-rest-client.version>
-        <microprofile-openapi.version>3.0</microprofile-openapi.version>
+        <microprofile-openapi.version>3.1</microprofile-openapi.version>
```
Comment (luiseufrasio) [inline pom.xml:218]: This is already done in FISH-6599

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/BaseProcessor.java (hunk 2 above)
Comment (aubi) [reply appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/BaseProcessor.java:81]: 3.1.0 breaks the MP TCK.

**Comment by breakponchito** [comment]
Jenkins test please

File: pom.xml (hunk 3 above)
Comment (aubi) [reply pom.xml:218]: git merge will handle it; without this, my PR wouldn't compile :-)

**Pull request merged**